Framework/PythonInterface/plugins/algorithms/ClipPeaks.py

```diff
@@ -0,0 +1,205 @@
+        weighted 4, i=+/- 1 weighted 3, i+/-2 weighted 2 and i+/-3
+        weighted 1 this input is only the y values. See the equation below:
+
+        Y_{i}^{s} = \\frac{ \\sum_{r=-r_{max}}^{r_{max}} (r_{max} - \\vert r \\vert ) \\cdot Y_{i+r} }
+                         { \\sum_{r=-r_{max}}^{r_{max}} (r_{max} - \\vert r \\vert ) }
+        where r_{max} is the "order" parameter (smoothing factor).
+
+        :param data: Input numpy.ndarray to smooth (Y data of the input workspace).
+        :param order: Smoothing factor, assumed an odd integer (even factors will be rounded to the next odd).
+        """
+        sm = np.zeros(len(data))
+        factor = int(order / 2) + 1
+
+        for i in range(len(data)):
+            temp = 0
+            ave = 0
+            for r in range(max(0, i - int(order / 2)), min(i + int(order / 2), len(data) - 1) + 1):
+                temp = temp + (factor - abs(r - i)) * data[r]
+                ave = ave + factor - abs(r - i)
+            sm[i] = temp / ave
+
+        return sm
+
+    def log_log_sqrt_transformation(self, input):
+        """
+        This transforms data to be more sensitive to weak peaks. The
+        function is reversed by the inv_log_log_sqrt_transformation function below
+        """
+        out = np.log(np.log((input + 1) ** 0.5 + 1) + 1)
+
+        return out
+
+    def inv_log_log_sqrt_transformation(self, input):
+        """
+        Applies the inverse log-log-sqrt function on the input data. See Function LLS function above
+        """
+        out = (np.exp(np.exp(input) - 1) - 1) ** 2 - 1
+
+        return out
+
+    def peak_clip(self, data, win=30, decrease=True, lls=True, smooth_window=0):
+        """
+        Performs peak clipping on the data iteratively over the size of win. The input data can be optionally
+        smoothed if smooth_window is greater than 0. The LLS function can be performed on the data after smoothing
+        to help exaggerate smaller peaks present in the data before clipping. The peak clipping window used for
+        iteration is done with a decreasing window unless decrease is False.
+
+        :param data: Input (Y) data to clip.
+        :param win: Window size of the clipping (controls the iteration count of the clipping function).
+        :param decrease: Flag to determine if an increasing/decreasing window is applied to the data.
+        :param lls: Flag to enable/disable applying the LLS function to data before clipping.
+        :param smooth_window: Smoothing factor for the smoothing function. 0 to apply no smoothing.
+        :return: np.ndarray of peak clipped data
+        """
+        start_data = np.copy(data)
+
+        window = win
+        self.log().information('Smoothing window: {0}'.format(str(smooth_window)))
+
+        if smooth_window > 0:
+            data = self.smooth(data, smooth_window)
+
+        if lls:
+            data = self.log_log_sqrt_transformation(data)
+
+        temp = data.copy()
```

Comment:
Is it necessary to make a copy of **data**, if **data** is not used in this method anymore?